cloudformation/OpenMapKitServer.template.js

```diff
@@ -0,0 +1,178 @@
+  },
+  ELBSecurityGroup: {
+    Description: 'Security Group for the ELB',
+    Type: 'String'
+  },
+  ELBSubnets: {
+    Description: 'ELB subnets',
+    Type: 'String'
+  },
+  EC2SecurityGroup: {
+    Description: 'EC2 security group',
+    Type: 'String'
+  },
+  S3Bucket: {
+    Description: 'S3 bucket',
+    Type: 'String'
+  }
+};
+
+const Resources = {
+  OpenMapKitServerASG: {
+    DependsOn: 'OpenMapKitServerLaunchConfiguration',
+    Type: 'AWS::AutoScaling::AutoScalingGroup',
+    Properties: {
+      AutoScalingGroupName: cf.stackName,
+      Cooldown: 300,
+      MinSize: 0,
+      DesiredCapacity: 1,
+      MaxSize: 1,
+      HealthCheckGracePeriod: 300,
+      LaunchConfigurationName: cf.stackName,
+      LoadBalancerNames: [ cf.ref('OpenMapKitServerLoadBalancer') ],
+      HealthCheckType: 'EC2',
+      AvailabilityZones: cf.getAzs(cf.region)
+    }
+  },
+  OpenMapKitServerScaleUp: {
+      Type: "AWS::AutoScaling::ScalingPolicy",
+      Properties: {
+        AutoScalingGroupName: cf.ref('OpenMapKitServerASG'),
+        PolicyType: 'TargetTrackingScaling',
+        TargetTrackingConfiguration: {
+          TargetValue: 85,
+          PredefinedMetricSpecification: {
+            PredefinedMetricType: 'ASGAverageCPUUtilization'
+          }
+        },
+        Cooldown: 300
+      }
+  },
+  OpenMapKitServerLaunchConfiguration: {
+    Type: 'AWS::AutoScaling::LaunchConfiguration',
+      Properties: {
+        IamInstanceProfile: cf.ref('OpenMapKitServerEC2InstanceProfile'),
+        ImageId: 'ami-0e4372c1860d7426c',
+        InstanceType: 'm3.large',
+        LaunchConfigurationName: cf.stackName,
+        SecurityGroups: [cf.ref('EC2SecurityGroup')],
+        UserData: cf.userData([
+          '#!/bin/bash',
+          'apt update -y \ ',
+          ' && apt upgrade -y \ ',
+          ' && apt install -y --no-install-recommends \ ',
+          '   apt-transport-https \ ',
+          '   curl \ ',
+          '   software-properties-common \ ',
+          ' && curl -sf https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \ ',
+          ' && add-apt-repository -y -u -s "deb https://deb.nodesource.com/node_6.x $(lsb_release -c -s) main" \ ',
+          ' && apt install -y --no-install-recommends \ ',
+          '   build-essential \ ',
+          '   default-jre-headless \ ',
+          '   git \ ',
+          '   nodejs \ ',
+          '   python \ ',
+          '   python-dev \ ',
+          '   python-pip \ ',
+          '   python-setuptools \ ',
+          '   python-wheel \ ',
+          ' && apt-get clean \ ',
+          '&& rm -rf /var/lib/apt/lists/* ',
+          'npm install -g yarn \ ',
+          '&& rm -rf /root/.npm',
+          'mkdir -p /app',
+          'git clone https://github.com/hotosm/OpenMapKitServer.git && cp -R OpenMapKitServer/* /app/ && cd app',
+          'pip install -r requirements.txt',
+          'yarn && rm -rf /root/.cache/yarn',
+          'yarn pushbuild',
```

Comment:
This command is to send the frontend build to git.
On the cfn, we need to pull the `dist` branch and put it on the `frontend/build/` folder

Reply:
👋 @willemarcel Thanks for reviewing. ♥️ 

I am not sure I follow ^ entirely - is it something like `git checkout dist && cp . -R frontend/build && git checkout master`

Reply:
My suggestion. Let's define a variable to the OpenMapKit version. I don't know where is more appropriate to set it on the cfn...

```
VERSION=v1.3.0
git clone https://github.com/hotosm/OpenMapKitServer.git /app/ && cd app && git checkout $VERSION
```

So we can download the frontend build and copy it to its proper directory with

```
wget https://github.com/hotosm/OpenMapKitServer/archive/${VERSION}-frontend.tar.gz -P /tmp/
rm frontend/build/* -R
tar -xvzf /tmp/${VERSION}-frontend.tar.gz -C frontend/build/ --strip 1
```